### PR TITLE
hook up contacts joined screen

### PR DESCRIPTION
**Pull request by jakob223**
image

TODO:
 - figure out waiting key situation on follow buttons
 - make the wave buttons not change to zero size while spinning

### CODE REVIEW

**Comment by jakob223** [comment]
oops forgot to CC @keybase/y2ksquad

File: shared/settings/contacts-joined/index.native.tsx [hunk 1]
```diff
@@ -2,41 +2,46 @@ import * as React from 'react'
...
-      <Kb.ConnectedUsernames colorFollowing={true} type="BodySemibold" usernames={[username]} />
-      <Kb.Text type="BodySmall">{contactLabel}</Kb.Text>
-      <Kb.Box2 direction="horizontal" gap="tiny" fullWidth={true} style={styles.buttons}>
-        <FollowButton username={username} small={true} />
-        <WaveButton usernames={username} small={true} />
+const renderItem = (_: number, item: RPCTypes.ProcessedContact) => {
+  const username = item.username
+  const label =
+    item.contactName || (item.component && (item.component.phoneNumber || item.component.email)) || ''
+  const followThem = Container.useSelector(state => Tracker2Constants.followThem(state, username))
```
Comment (buoyad) [inline shared/settings/contacts-joined/index.native.tsx]: This isn't a component, can't call hooks here. I get a red screen on enabling contact import.

File: shared/reducers/settings.tsx [hunk 2]
```diff
@@ -202,6 +202,15 @@ function reducer(state: Types.State = initialState, action: Actions): Types.Stat
...
       )
+    case SettingsGen.showContactsJoinedModal:
+      return state.update('contacts', contacts =>
+        contacts.merge({
+          alreadyOnKeybase: I.List(action.payload.newlyResolved),
+          waitingToShowJoinedModal: false,
+        })
+      )
+    case SettingsGen.editContactImportEnabled:
+      return state.update('contacts', contacts => contacts.set('waitingToShowJoinedModal', true))
```
Comment (buoyad) [inline shared/reducers/settings.tsx]: Only do this if `fromSettings`?

File: shared/teams/invite-by-contact/fetch-contacts.native.tsx [hunk 3]
```diff
@@ -108,7 +108,7 @@ const WithContacts = (props: WithContactsProps) => {
     // whether to dispatch `createRequestContactPermissions` so we never
     // dispatch more than once.
     if (permStatus === 'unknown' || permStatus === 'undetermined') {
-      dispatch(SettingsGen.createRequestContactPermissions({thenToggleImportOn: false}))
+      dispatch(SettingsGen.createRequestContactPermissions({fromSettings: false, thenToggleImportOn: false}))
```
Comment (buoyad) [inline shared/teams/invite-by-contact/fetch-contacts.native.tsx]: :sheep: you could make `fromSettings` optional

File: go/contacts/phonebook.go [hunk 4]
```diff
@@ -70,10 +70,17 @@ type assertionToNameCache struct {
...
 	resolveResults, err := ResolveContacts(mctx, provider, contacts)
 	if err != nil {
-		return nil, err
+		return res, err
+	}
+
+	// find resolved contacts
+	for _, contact := range resolveResults {
+		if contact.Resolved {
+			res.Resolved = append(res.Resolved, contact)
```
Comment (buoyad) [inline go/contacts/phonebook.go]: can we filter out ones that are the current user?

File: shared/actions/chat2/index.tsx [hunk 5]
```diff
@@ -3264,6 +3264,18 @@ const gregorPushState = (state: TypedState, action: GregorGen.PushStatePayload,
   const isSearchNew = !items.some(i => i.item.category === Constants.inboxSearchNewKey)
   actions.push(Chat2Gen.createSetInboxShowIsNew({isNew: isSearchNew}))
 
+  const blockButtons = items.some(i => i.item.category.startsWith('blockButtons'))
+  if (blockButtons) {
+    const teamIDs = items
+      .filter(i => i.item.category.startsWith('blockButtons'))
+      .map(i => i.item.category.substr(13)) as Array<RPCTypes.TeamID>
```
Comment (buoyad) [inline shared/actions/chat2/index.tsx]: is it `blockButtons:`? can we put the prefix in constants

File: shared/actions/json/settings.json [hunk 6]
```diff
@@ -67,10 +67,15 @@
...
+      "thenToggleImportOn?": "boolean",
+      "fromSettings?": "boolean"
     },
     "editContactImportEnabled": {
-      "enable": "boolean"
+      "enable": "boolean",
+      "fromSettings?": "boolean"
+    },
+    "showContactsJoinedModal": {
+      "newlyResolved": "Array<RPCTypes.ProcessedContact>"
```
Comment (buoyad) [inline shared/actions/json/settings.json]: Confusing naming since this is taking `resolved` not `newlyResolved`

**Comment by jakob223** [comment]
Merging over failed test that fails in master

**Pull request merged**